packages/@aws-cdk/aws-eks/lib/service-account.ts

```diff
@@ -0,0 +1,74 @@
+import { FederatedPrincipal, IRole, OpenIdConnectProvider, Role } from '@aws-cdk/aws-iam';
+import { Construct } from '@aws-cdk/core';
+import { Cluster } from './cluster';
+
+/**
+ * Service Account
+ */
+export interface ServiceAccountOptions {
+  /**
+   * The cluster to apply the patch to.
+   */
+  readonly name: string;
+  /**
+   * The cluster to apply the patch to.
+   */
+  readonly namespace: string;
+}
+
+/**
+ * Service Account
+ */
+export interface ServiceAccountProps extends ServiceAccountOptions {
+  /**
+   * The cluster to apply the patch to.
+   * [disable-awslint:ref-via-interface]
+   */
+  readonly cluster: Cluster;
+}
+
+/**
+ * Service Account
+ */
+export class ServiceAccount extends Construct {
+
+  /**
+   * The role the service account is linked to.
+   */
+  public readonly role: IRole;
+
+  constructor(scope: Construct, id: string, props: ServiceAccountProps) {
+    super(scope, id);
+
+    const { cluster, name, namespace } = props;
+
+    let provider = cluster.node.tryFindChild('OpenIdConnectProvider') as OpenIdConnectProvider;
```

Comment:
I rather move this to the Cluster class as `cluster.openIdConnectProvider` as a lazy-initialized property:

```ts
class Cluster {
  public get openIdConnectProvider() {
    if (!this._openIdConnectProvider) {
      if (!this.kubectlEnabled) { throw new Error(`...`); }
      this._openIdConnectProvider = new OpenIdConnectProvider(this, 'OpenIdConnectProvider', {
        url: this.clusterOpenIdConnectIssuerUrl
      }
    }
    return this._openIdConnectProvider;
  }
}
```